documentation/decisions/adr/2024-03-19-dashboard-storage.md

```diff
@@ -0,0 +1,87 @@
+# Dashboard Data Storage
+
+- **Status:** Active
+- **Last Modified:** 2024-03-19
+- **Related Issue:** [#{issue number}](https://github.com/HHS/simpler-grants-gov/issues/{issue number})
+- **Deciders:** Aaron Couch, Billy Daly
+
+## Context and Problem Statement
+
+We want to determine the data storage solution for our upcoming business intelligence dashboard project.
+
+This data storage solution will be our canonical source of truth for many types of imported data. These data types will range from thousands of records a month (ex. sprint data), up to potentially millions of records of records (ex. Google Analytics) once this project reaches maturity. The chosen solution should be easy to use, cost-effective, and performant for all of our in-scope data types. The types of data we will be importing include:
+
+- Google Analytics
+- Grants program data, eg.
+  - Grants opportunities from the database
+  - Grant applications from S3
+- USA spending data CSVs
+- Communication platform stats, eg.
+  - Slack stats
+  - Google Group stats
+  - GitHub stats
+- API application metrics
+- API infrastructure metrics from Cloudwatch
+
+We will not be importing all of these types of data immediately. On the 0 - 6 month timeframe, we will only be importing the smaller datasets (thousands of records). By 2 - 5 years we will be importing all of these types of data, and our data size will be quite large (many millions of records). The desired solutions have different cost/performance characteristics in those time ranges, and we will need to evaluate those differences.
+
+## Decision Drivers <!-- RECOMMENDED -->
+
+- cost
+- ease of use
+- performance
+
+All of these options should be evaluated for 0 - 6 months, 6 months - 2 years, 2 years - 5 years
+
+## Options Considered
+
+- S3
+- Redshift
+- Postgres
+- Snowflake
```

Comment:
Would it be worth adding Aurora to this list of options per @bretthrosenblatt's comments in Slack?

Reply:
Aurora=Postgres in this respect

Reply:
I'd recommend both, actually

Reply:
Gotcha, I wasn't sure if postgres meant RDS -- that's what we're using for our main database right?

Reply:
@widal001 AWS RDS Aurora PostgreSQL is its full name, and its our main database yes

Reply:
> Gotcha, I wasn't sure if postgres meant RDS -- that's what we're using for our main database right?

Purposely confusing. There is RDS Postgres, which is cloud-hosted Postgres, and Postgres-compatible Aurora, which is a Postgres-like interface on an Aurora db engine and storage cluster (what you're using)